narwhals/_dask/expr.py

```diff
@@ -641,9 +674,65 @@ def func(df: DaskLazyFrame) -> list[Any]:
             root_names=self._root_names,
             output_names=self._output_names,
             returns_scalar=False,
+            modifies_index=False,
             backend_version=self._backend_version,
         )
 
+    def sort(self: Self, *, descending: bool = False, nulls_last: bool = False) -> Self:
+        na_position = "last" if nulls_last else "first"
+
+        def func(_input: Any, ascending: bool, na_position: bool) -> Any:  # noqa: FBT001
+            import dask_expr as de  # ignore-banned-import
+
+            name = _input.name
+
+            result = _input.to_frame(name=name).sort_values(
+                by=name, ascending=ascending, na_position=na_position
+            )[name]
+            return de._collection.Series(de._expr.AssignIndex(result, _input.index))
```

Comment:
Yeah I can second @MarcoGorelli here, please don't do this. We are sorting a Series (i.e. a single column from the df), correct?

I would:

```
tmp = _input.reset_index().sort_values(...)
result = tmp[_input.name]
result.index = tmp["use the index name of _input"]
```

Reply:
Or do you want to keep the Index of the input?

If yes, this is fundamentally a bad idea in Dask, it will shoot you in the foot all over the place. You have zero guarantees that the partitions keep their lengths when sorting (it is a lot more likely that they do not), so this is bound to fail in all kinds of places

Reply:
Hey @phofl thanks for taking the time.

> We are sorting a Series (i.e. a single column from the df), correct?

Yes indeed, but with the final goal to potentially add it as a new column to the original dataframe, and that's where the index misalignment gets into the game.

I will try the approach you are suggesting, which is not too far off from what is already implemented, and see if everything else falls into place 🙏🏼